File: app/src/main/java/com/ajithvgiri/phoneval/utils/PreferenceUtils.java
package com.ajithvgiri.phoneval.utils;

import android.content.Context;
import android.content.SharedPreferences;

import static android.content.Context.MODE_PRIVATE;

public class PreferenceUtils {

    private static final String PREFS_FILE_NAME = "QkopyPreference";

    public static void firstTimeAskingPermission(Context context, String permission, boolean isFirstTime) {
        SharedPreferences sharedPreference = context.getSharedPreferences(PREFS_FILE_NAME, MODE_PRIVATE);
        sharedPreference.edit().putBoolean(permission, isFirstTime).apply();
    }

    public static boolean isFirstTimeAskingPermission(Context context, String permission) {
        return context.getSharedPreferences(PREFS_FILE_NAME, MODE_PRIVATE).getBoolean(permission, true);
    }
}

File: app/src/main/java/com/ajithvgiri/phoneval/utils/PreferenceUtils.kt
package com.ajithvgiri.phoneval.utils

import android.content.Context
import android.content.Context.MODE_PRIVATE

object PreferenceUtils {

    private val PREFS_FILE_NAME = "QkopyPreference"

    fun firstTimeAskingPermission(context: Context, permission: String, isFirstTime: Boolean) {
        val sharedPreference = context.getSharedPreferences(PREFS_FILE_NAME, MODE_PRIVATE)
        sharedPreference.edit().putBoolean(permission, isFirstTime).apply()
    }

    fun isFirstTimeAskingPermission(context: Context, permission: String): Boolean {
        return context.getSharedPreferences(PREFS_FILE_NAME, MODE_PRIVATE).getBoolean(permission, true)
    }
}
